components/feature/intent/src/main/java/mozilla/components/feature/intent/processing/TabIntentProcessor.kt

```diff
@@ -73,6 +77,37 @@ class TabIntentProcessor(
         }
     }
 
+    private fun processSearchIntent(intent: SafeIntent): Boolean {
+        val searchQuery = intent.getStringExtra(SearchManager.QUERY)
+
+        return if (searchQuery.isNullOrBlank()) {
+            false
+        } else {
+            resolveSearch(searchQuery, Source.ACTION_SEARCH)
+            true
+        }
+    }
+
+    private fun processWebSearchIntent(intent: SafeIntent): Boolean {
+        val searchQuery = intent.getStringExtra(SearchManager.QUERY)
+
+        return if (searchQuery.isNullOrBlank()) {
+            false
+        } else {
+            resolveSearch(searchQuery, Source.ACTION_WEB_SEARCH)
+            true
```

Comment:
I think it's fine to just have a single `Source.ACTION_SEARCH` to represent both search intents. Then you can collapse the process search intent functionality into a single function rather than 3.

Reply:
Done! :)